src/components/app/TabBar.js

```diff
@@ -51,6 +52,21 @@ class TabBar extends PureComponent {
   }
 }
 
+type Props = {
+  className?: string,
+  tabs: Tab[],
+  selectedTabName: string,
+  tabOrder: number[],
+  onSelectTab: string => mixed,
+  onChangeTabOrder: (number[]) => any,
+  +_mouseDownListener?: () => mixed,
+};
+
+type Tab = {
```

Comment:
Types in Flow are hoisted, but I generally prefer them to be in their order of use, so here Tab would be above Props, and Props would be above the Component. PropTypes would continue to be at the bottom.